### PR TITLE
CLOUDP-164770 Add e2e tests for "serverless backups" commands

### PR DESCRIPTION
**Pull request by fmenezes**
## Proposed changes

Add e2e tests for "serverless backups" commands

_Jira ticket:_ CLOUDP-164770

## Checklist

- [ ] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added any necessary documentation in document requirements section listed in [CONTRIBUTING.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/CONTRIBUTING.md) (if appropriate)
- [ ] I have addressed the @mongodb/docs-cloud-team comments (if appropriate)
- [ ] I have updated [test/README.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/test/README.md) (if an e2e test has been added)
- [ ] I have run `make fmt` and formatted my code

## Further comments

- Adding tests
- Added validation for delivery type on `atlas serverless backup restore create`
- Made backup tests run in parallel

### CODE REVIEW

File: test/e2e/atlas/backup_serverless_test.go [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+	g.generateServerlessCluster()
+
+	t.Run("Snapshot List", func(t *testing.T) {
+		cmd := exec.Command(cliPath,
+			serverlessEntity,
+			backupsEntity,
+			snapshotsEntity,
+			"list",
+			clusterName,
+			"-o=json")
```
Comment (andreaangiolillo) [inline test/e2e/atlas/backup_serverless_test.go:51]: Don't you need to pass the `--projectId` flag to use the project that you created at line 41?

**Review by andreaangiolillo — APPROVED** [review]
LGTM

File: test/e2e/atlas/backup_serverless_test.go (hunk 1 above)
Comment (fmenezes) [reply test/e2e/atlas/backup_serverless_test.go:51]: no, the clusterName here is from a cluster that lives on the main `CLI E2E` project

**Review by tibulca — APPROVED** [review]

File: test/e2e/atlas/backup_serverless_test.go (hunk 1 above)
Comment (andreaangiolillo) [reply test/e2e/atlas/backup_serverless_test.go:51]: Thanks!

**Review by blva — DISMISSED** [review]
LGTM
[q] did we add to the `.md` file? OK to add later w/ all the e2e tests as well

**Pull request merged**